okhttp/src/main/kotlin/okhttp3/internal/http2/Http2Reader.kt

```diff
@@ -146,7 +146,10 @@ class Http2Reader(
     headerBlockLength = lengthWithoutPadding(headerBlockLength, flags, padding)
     val headerBlock = readHeaderBlock(headerBlockLength, padding, flags, streamId)
 
-    handler.headers(endStream, streamId, -1, headerBlock)
+    // Ignore early hints
+	  if (!(headerBlockLength != 0 && headerBlock[0].value.equals(EARLY_HINTS_CODE))) {
```

Comment:
It doesn't seem like ":status" has to be the first header. Will just check it's the ":status" but otherwise leave as is.

Reply:
Mentioned in description.